Core/src/org/sleuthkit/autopsy/datasourcesummary/datamodel/DataSourceUserActivitySummary.java

```diff
@@ -0,0 +1,439 @@
+        return IntStream.range(0, count)
+                .mapToObj(num -> new TopDomainsResult(
+                getId.apply("domain", num),
+                getId.apply("url", num),
+                (long) num,
+                new Date(((long) num) * 1000 * 60 * 60 * 24)
+        ))
+                .collect(Collectors.toList());
+    }
+
+    private final SleuthkitCaseProvider caseProvider;
+    private final TextTranslationService translationService;
+    private final java.util.logging.Logger logger;
+
+    /**
+     * Main constructor.
+     */
+    public DataSourceUserActivitySummary() {
+        this(SleuthkitCaseProvider.DEFAULT, TextTranslationService.getInstance(),
+                org.sleuthkit.autopsy.coreutils.Logger.getLogger(DataSourceUserActivitySummary.class.getName()));
+    }
+
+    /**
+     * Main constructor with external dependencies specified. This constructor
+     * is designed with unit testing in mind since mocked dependencies can be
+     * utilized.
+     *
+     * @param provider           The object providing the current SleuthkitCase.
+     * @param translationService The translation service.
+     * @param logger             The logger to use.
+     */
+    public DataSourceUserActivitySummary(
+            SleuthkitCaseProvider provider,
+            TextTranslationService translationService,
+            java.util.logging.Logger logger) {
+
+        this.caseProvider = provider;
+        this.translationService = translationService;
+        this.logger = logger;
+    }
+
+    /**
+     * Throws an IllegalArgumentException if count <= 0.
+     *
+     * @param count The count being checked.
+     */
+    private void assertValidCount(int count) {
+        if (count <= 0) {
+            throw new IllegalArgumentException("Count must be greater than 0");
+        }
+    }
+
+    /**
+     * Retrieves most recent web searches by most recent date grouped by search
+     * term.
+     *
+     * @param dataSource The data source.
+     * @param count      The maximum number of records to be shown (must be >
+     *                   0).
+     *
+     * @return The list of most recent web searches where most recent search
+     *         appears first.
+     *
+     * @throws
+     * org.sleuthkit.autopsy.datasourcesummary.datamodel.SleuthkitCaseProvider.SleuthkitCaseProviderException
+     * @throws TskCoreException
+     */
+    public List<TopWebSearchResult> getMostRecentWebSearches(DataSource dataSource, int count) throws SleuthkitCaseProviderException, TskCoreException {
+        assertValidCount(count);
+
+        List<TopWebSearchResult> results = caseProvider.get().getBlackboard().getArtifacts(ARTIFACT_TYPE.TSK_WEB_SEARCH_QUERY.getTypeID(), dataSource.getId())
+                .stream()
+                // get items where search string and date is not null
+                .map(artifact -> {
```

Comment:
This would be very difficult to debug if there was an issue.  You might want to consider revising.

Reply:
Ok.  I'll break it up a little bit.